### ISSUE TITLE
bitmaptools.boundary_fill() should do background processing

### ISSUE THREAD

**Issue by dhalbert** [adafruit/circuitpython#7485]
I landed here after having some freeze problems. Just keeping this here for future reference and in case somebody gets to the same point as myself.
When I tried the following modified code,  the communication was lost and the editor (MU) and became unresponsive. 
I did the same using TIO and Screen command in ubuntu causing disconnection problems. After a while the code will show the expected result, however the communication is lost with the board. This is a little problem, but if people is impatient as I am, the board will get in safe mode very easily :)

```python
import board
import displayio
import bitmaptools

display = board.DISPLAY

bitmap = displayio.Bitmap(display.width, display.height, 3)

palette = displayio.Palette(3)
palette[0] = 0x000000
palette[1] = 0xffffff
palette[2] = 0x0000ff

tile_grid = displayio.TileGrid(bitmap, pixel_shader=palette)
group = displayio.Group()
group.append(tile_grid)
display.show(group)

# Draw a rectangle
for x in range(150, 170):
    bitmap[x, 30] = 1
    bitmap[x, 50] = 1
for y in range(30, 51):
    bitmap[150, y] = 1
    bitmap[170, y] = 1

# Draw a non-rectangle
for i in range(30, 120):
    bitmap[i, i] = 1
    bitmap[i+30, i] = 1

for x in range(0, 30):
    bitmap[x+30, 30] = 1
    bitmap[x+120, 120] = 1

# fill rectangle white
bitmaptools.boundary_fill(bitmap,display.width//2, display.height//2, 1, 0)

# fill non-rectangle blue
bitmaptools.boundary_fill(bitmap,40, 35, 2, 0)

while True:
    pass
```

### RESULTS

* Screen
```
home@home:~$ screen /dev/ttyACM0 115200
[screen is terminating]
home@home:~$ 
```

* TIO
```
Adafruit CircuitPython 8.0.0-beta.6 on 2022-12-21; Adafruit PyPortal Titano with samd51j20
>>> import q
[tio 19:51:52] Disconnected
[tio 19:52:33] Connected
🐍REPL | 8.0.0-beta.6

```

Tested on `Adafruit CircuitPython 8.0.0-beta.6 on 2022-12-21; Adafruit PyPortal Titano with samd51j20`

_Originally posted by @jposada202020 in https://github.com/adafruit/circuitpython/issues/5145#issuecomment-1401227189_

**Comment by dhalbert** [comment]
from https://github.com/adafruit/circuitpython/pull/5145#issuecomment-1404123840 (@tannewt):
 
> Sounds like the `boundary_fill` loop needs a `RUN_BACKGROUND_TASKS`; to handle USB while it fills.

**Comment by FoamyGuy** [comment]
Is there anything beyond `RUN_BACKGROUND_TASKS;` that is needed, or could be used inside of the main while loop in that function to allow it to bail early if/when the user does ctrl-c?

I added `RUN_BACKGROUND_TASKS;` here:

https://github.com/adafruit/circuitpython/blob/0ca6cc7741103ced6e2c35fb3d7da708fb030d7f/shared-module/bitmaptools/__init__.c#L377-L378

That does allow the device not to disconnect while the function is running.

However I noticed if I press ctrl-c while it's running there is still a very long delay until the keyboard interrupt actually gets raised. It seems like the function finishes all of it's work (which takes a rather long time for large bitmaps) before the keyboard interrupt gets raised
